libraries/standard/mqtt/integration-test/mqtt_system_test.c

```diff
@@ -687,9 +731,11 @@ void test_MQTT_Subscribe_Publish_With_Qos_2( void )
     /* We expect PUBREC and PUBCOMP responses for the PUBLISH request, and
      * incoming PUBLISH with the same message that we published (as we are subscribed
      * to the same topic). Also, we expect a PUBREL ack response from the server for
-     * the incoming PUBLISH (as we subscribed and publish with QoS 2). */
+     * the incoming PUBLISH (as we subscribed and publish with QoS 2). Since it takes
+     * longer to complete a QoS 2 publish, we run the process loop longer to allow it
+     * ample time. */
     TEST_ASSERT_EQUAL( MQTTSuccess,
-                       MQTT_ProcessLoop( &context, MQTT_PROCESS_LOOP_TIMEOUT_MS ) );
+                       MQTT_ProcessLoop( &context, 2 * MQTT_PROCESS_LOOP_TIMEOUT_MS ) );
```

Comment:
Was use of 1400 ms required on your local setup for this test?

Reply:
Yes

Reply:
I'm concerned that increasing the timeout could would increase the time for CI tests (when they are enabled), and it will go unnoticed.
For my EC2 machine, 700ms worked fine. (As the CI tests would run on EC2 isntances, benchmarking the timing against that should be fine)

Reply:
I don't think we should allow tests to fail on our local setups